### DISCUSSION TITLE
Can flux_bootstrap_git terraform provider support git branch variable?

### DISCUSSION THREAD

**Discussion by tsiq-sanjanar** [fluxcd/flux2#4740] [Q&A] [answered]
Hi,

For our usecase we would like to have each cluster flux components under separate git branch. Does flux2 only support main branch? 

There is no branch variable here https://registry.terraform.io/providers/fluxcd/flux/latest/docs/resources/bootstrap_git

for the below bootstrap  resource
```
resource "flux_bootstrap_git" "bootstrap" {
  version                = var.flux2_version
  path                   = "clusters/${var.cluster_name}"
  branch                 = var.flux2_branch 
  kustomization_override = file("${path.module}/kustomization.yaml")
  components_extra       = ["image-reflector-controller", "image-automation-controller"]
}
```
I get below error 
```
│ Error: Unsupported argument
│ 
│   on flux2.tf line 22, in resource "flux_bootstrap_git" "bootstrap":
│   22:   branch                 = var.flux2_branch
│ 
│ An argument named "branch" is not expected here.
```
Thank you.

**Comment by stefanprodan** [comment, marked as answer]
The branch is a property of Git, where you configured your Git URL. https://registry.terraform.io/providers/fluxcd/flux/latest/docs#nested-schema-for-git

**Comment by tsiq-sanjanar** [comment]
Thank you sorry about that new to flux 2
